drivers/ISOSR.py

```diff
@@ -546,13 +546,12 @@ def appendCIFSMountOptions(self, mountcmd):
 
             if not cifutils.containsCredentials(self.dconf, prefix="cifs"):
                 options.append('guest')
+                domain = None
+            else:
+                _, domain = cifutils.splitDomainAndUsername(self.dconf['username'])
 
             options.append(self.getSMBVersion())
 
-            username, domain = (
-                cifutils.splitDomainAndUsername(self.dconf['username'])
-            )
-
             if domain:
```

Comment:
Why not just move this up into the `else` clause ?

Reply:
That's what I thought I did. Is the diff somehow showing back-to-front?

Reply:
> ~That's what I thought I did. Is the diff somehow showing back-to-front?~ Oh, you're talking about the `if domain`. No, that has to be kept, because domain can come back `None` from `splitDomainAndUsername`.

Well, yes, but we know that there can never be a domain in the positive case of the `if` so making the `else` be
``` python
else: 
    _, domain = cifutils.splitDomainAndUsername(self.dconf['username'])
    if domain:
         options.append('domain=' + domain)
```
would then remove the need for the `domain=None` in the guest handling side of the branch.

Maybe we should do this as a future improvement to avoid the sensitivity of the UTs to the order of the provided options.